### PR TITLE
Add Inverse_normal_cdf Presto Function

### PR DESCRIPTION
**Pull request by rrando901**
Add Presto function inverse_normal_cdf following the Java implementation([source](https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java#L803))

Resolves: https://github.com/facebookincubator/velox/issues/5476

### CODE REVIEW

File: velox/functions/prestosql/Probability.h [hunk 1]
```diff
@@ -91,5 +92,17 @@ struct BinomialCDFFunction {
...
 
+template <typename T>
+struct InverseNormalCDFFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(double& result, double m, double sd, double p) {
+    VELOX_USER_CHECK((p >= 0) && (p <= 1), "p must be 0 > p > 1");
+    VELOX_USER_CHECK_GT(sd, 0, "standardDeviation must be > 0");
+
+    result = m + sd * 1.4142135623730951 * boost::math::erf_inv(2 * p - 1);
```
Comment (aditi-pandit) [inline velox/functions/prestosql/Probability.h:103]: 1.4.14 is square root of 2. Please add a constant variable for it, and use it in the formula.

File: velox/functions/prestosql/tests/ProbabilityTest.cpp [hunk 2]
```diff
@@ -155,5 +155,43 @@ TEST_F(ProbabilityTest, binomialCDF) {
...
+  EXPECT_THAT(inverseNormalCDF(kNan, 2, 0.1985), IsNan());
+
+  EXPECT_EQ(inverseNormalCDF(std::nullopt, 1, 1), std::nullopt);
+  EXPECT_EQ(inverseNormalCDF(1, 1, std::nullopt), std::nullopt);
+  EXPECT_EQ(inverseNormalCDF(std::nullopt, 1, std::nullopt), std::nullopt);
+  EXPECT_EQ(
+      inverseNormalCDF(std::nullopt, std::nullopt, std::nullopt), std::nullopt);
+
+  // Test invalid inputs.
+  VELOX_ASSERT_THROW(
```
Comment (aditi-pandit) [inline velox/functions/prestosql/tests/ProbabilityTest.cpp:183]: Please add few tests with kInf values also.

**Review by aditi-pandit — COMMENTED** [review]
Thanks for adding this function.

File: velox/functions/prestosql/tests/ProbabilityTest.cpp (hunk 2 above)
Comment (rrando901) [reply velox/functions/prestosql/tests/ProbabilityTest.cpp:187]: Added.

**Comment by rrando901** [comment]
Fuzzer results

```
I20230630 13:18:33.328297 845386 ExpressionVerifier.cpp:56] ===================
I20230630 13:18:33.328333 845386 ExpressionVerifier.cpp:67] All results match.
I20230630 13:18:33.328408 845386 ExpressionFuzzer.cpp:1357] ==============================> Done with iteration 411
I20230630 13:18:33.328904 845386 ExpressionFuzzer.cpp:1092] ==============================> Top 1 by number of rows processed
I20230630 13:18:33.328928 845386 ExpressionFuzzer.cpp:1094] Format: functionName numTimesSelected proportionOfTimesSelected numProcessedRows
I20230630 13:18:33.328948 845386 ExpressionFuzzer.cpp:1098] inverse_normal_cdf 43268 100.00% 4947
I20230630 13:18:33.328974 845386 ExpressionFuzzer.cpp:1104] ==============================> Bottom 1 by number of rows processed
I20230630 13:18:33.328992 845386 ExpressionFuzzer.cpp:1106] Format: functionName numTimesSelected proportionOfTimesSelected numProcessedRows
I20230630 13:18:33.329011 845386 ExpressionFuzzer.cpp:1111] inverse_normal_cdf 43268 100.00% 4947
I20230630 13:18:33.329033 845386 ExpressionFuzzer.cpp:1124] ==============================> All stats sorted by number of times the function was chosen
I20230630 13:18:33.329053 845386 ExpressionFuzzer.cpp:1126] Format: functionName numTimesSelected proportionOfTimesSelected numProcessedRows
I20230630 13:18:33.329073 845386 ExpressionFuzzer.cpp:1130] inverse_normal_cdf 43268 100.00% 4947
[==========] Running 0 tests from 0 test suites.
[==========] 0 tests from 0 test suites ran. (0 ms total)
[  PASSED  ] 0 tests.
```

**Review by aditi-pandit — APPROVED** [review]

**Comment by rrando901** [comment]
@aditi-pandit @amitkdutta @Yuhta 
I've opened an a new PR with these changes that is up-to-date with the main branch - https://github.com/facebookincubator/velox/pull/5797

**Pull request closed without merging**